site/src/Consumer/UpdateGitHubMetadataConsumer.php

```diff
@@ -0,0 +1,247 @@
+use Exception;
+use Github\Client;
+use Github\Exception\ErrorException;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Enrich metadata from GitHub
+ *
+ * This implementation uses GitHub GraphQL API to fetch data from GitHub API
+ *
+ * @author Amitosh Swain Mahapatra <amitosh.swain@gmail.com>
+ */
+class UpdateGitHubMetadataConsumer implements ConsumerInterface
+{
+
+    private const GH_REGEX =
+        '/^https:\/\/github\.com\/([^\/]+)\/([^\/]+?)(?:\.git)?$/';
+
+    /**
+     * @var ProjectRepository
+     */
+    private $repository;
+
+    /**
+     * @var GitHubApiService
+     */
+    private $githubApiService;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * UpdateGithubMetadataConsumer constructor.
+     *
+     * @param ProjectRepository      $repository
+     * @param GitHubApiService       $githubApiService
+     * @param EntityManagerInterface $entityManager
+     * @param LoggerInterface        $logger
+     */
+    public function __construct(
+        ProjectRepository $repository,
+        GitHubApiService $githubApiService,
+        EntityManagerInterface $entityManager,
+        LoggerInterface $logger
+    ) {
+
+        $this->repository = $repository;
+        $this->githubApiService = $githubApiService;
+        $this->entityManager = $entityManager;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param AMQPMessage $msg The message
+     *
+     * @return mixed false to reject and requeue, any other value to acknowledge
+     */
+    public function execute(AMQPMessage $msg)
+    {
+        try {
+            $projectId = (int) unserialize($msg->body);
+            /** @var Project $project */
+            $project = $this->repository->find($projectId);
+
+            if (!$project) {
+                // this should not happen in production
+                // but happens in dev if for some reason we clear the projects
+                // table
+                $this->logger->error(
+                    "Unable to update project with ID $projectId: project does "
+                    ."not exist."
+                );
+
+                return true; // don't requeue
+            }
+
+            $this->updateGithubMetadata($project);
+
+            return true;
+        } catch (Exception $e) {
+            // We got an unexpected Exception. We assume this is a one-off event
+            // and just log it, but otherwise keep the consumer running for the
+            // next requests.
+            $this->logger->error(
+                "Processing of repository resulted in an ".get_class($e)
+            );
+            $this->logger->error('Message: '.$e->getMessage());
+            $this->logger->error('Trace: '.$e->getTraceAsString());
+
+            return false;
+        }
+    }
```

Comment:
Most code in this consumer is copied from UpdateProjectInformation. Can you make a common base class for these two, e.g. `ProjectConsumer` that contains the shared code in `execute()`, and calls a `executeProjectUpdate(Project)` function or something similar?